validation_code/ConditionalLuminosityFunction.py

```diff
@@ -18,21 +19,27 @@ def __init__(self, band='r', magnitude_bins=None, mass_bins=None, z_bins=None, *
                                'Mag_true_{}_sdss_z01',
                               )
 
-        self.possible_Mag_fields = [f.format(band) for f in possible_Mag_fields]
-        self.band = band
+        self.possible_Mag1_fields = [f.format(band1) for f in possible_Mag_fields]
+        self.possible_Mag2_fields = [f.format(band2) for f in possible_Mag_fields]
+        self.band1 = band1
+        self.band2 = band2
+        self.color_cut_percent = float(color_cut_percent)
 
-        self.magnitude_bins   = magnitude_bins or np.linspace(-25, -18, 29)
-        self.mass_bins        = mass_bins or np.logspace(12, 15, 5)
-        self.z_bins           = z_bins or np.linspace(0, 0.5, 3)
+        self.magnitude_bins   = magnitude_bins or np.linspace(-26, -18, 29)
+        self.mass_bins        = mass_bins or np.logspace(13.5, 15, 5)
+        self.z_bins           = z_bins or np.linspace(0.2, 1.0, 4)
 
         self.n_magnitude_bins = len(self.magnitude_bins) - 1
         self.n_mass_bins      = len(self.mass_bins) - 1
         self.n_z_bins         = len(self.z_bins) - 1
 
         self.dmag = self.magnitude_bins[1:] - self.magnitude_bins[:-1]
         self.mag_center = (self.magnitude_bins[1:] + self.magnitude_bins[:-1])*0.5
+        
+        self.color_cut = lambda g, r, z: g-r>np.sort((g-r)[z<0.2])[-int(len(g[z<0.2])*self.color_cut_percent)]
```

Comment:
Use `np.percentile` so it's more readable: `g-r > np.percentile((g-r)[z < 0.2], self.color_cut_percent*100)`